### DISCUSSION TITLE
ERROR webdriver: TypeError: Invalid URL - ERROR @wdio/runner: Error: Failed to create session. - Help!

### DISCUSSION THREAD

**Discussion by sperezc06** [webdriverio/webdriverio#13509] [Q&A]
Hello, 

please see the error that I'm receiving trying to run npx wdio run wdio.conf.js. I have already run all the configuration with npm init wdio . 

![image](https://github.com/user-attachments/assets/2f579f7f-df77-45d7-a140-0f9881d3f880)

Pleae see my wdio.conf.ts 

![image](https://github.com/user-attachments/assets/7f4af871-9974-40f2-939f-0546fd4f73dc)
![image](https://github.com/user-attachments/assets/3d38b209-2f9d-4c07-a83f-e5a29334f976)
![image](https://github.com/user-attachments/assets/27fd7c01-eb01-4f0c-a660-f8bdf2bb4ff3)
![image](https://github.com/user-attachments/assets/d95459a0-1189-4d6b-a995-3fa2b4ab8424)

**Comment by christian-bromann** [comment]
@sperezc06 thanks for raising the issue. Mind creating a minimal reproducible example as GitHub project for us to check out? Thanks!

**Comment by renan2158** [comment]
I'm facing the same error. When I provide `appium:appWaitActivity` at the device capabilities, the error goes away but I get stuck by another one.

**Comment by renan2158** [comment]
@christian-bromann I know I didn't open the discussion and without wanting to take the lead, but looking at @sperezc06 it looks like to be the same problem I'm stuck right now. You finding out the solution would help me a lot because I've been stuck on it for a while. So, I created this [repo](https://github.com/renan2158/wdio-device-running-error) with the same project I have locally.

**Reply by christian-bromann** [reply]
Thanks for providing a reproduction case. I can't promise I will be able to look into it anytime soon especially since I am not super proficient in mobile testing.

**Reply by renan2158** [reply]
No problem, thank you anyway!

**Comment by renan2158** [comment]
@sperezc06 I just solved the problem by installing an older version of Webdriverio. I did it executing `npm init wdio@8.4.7 <path> --npm-tag=8.40.3`

**Comment by azerum** [comment]
In my case the issue was resolved when I explicitly set `port` option in `wdio.conf.ts` (or `wdio.conf.js`)

Note for people who can fix this: Looking at my stack trace, the error comes from [this line](https://github.com/webdriverio/webdriverio/blob/74346f930b035a475733748974dcec8893c08865/packages/webdriver/src/request/index.ts#L111). When `port` is not set, the code tries to parse URL `http://localhost:undefined/session` - URL with invalid port
